ENIGMAsystem/SHELL/Collision_Systems/Polygon/Polygonimpl.cpp

```diff
@@ -0,0 +1,710 @@
+/** Copyright (C) 2021 Nabeel Danish
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+////////////////////////////////////
+// Collision implementation functions - API dependant implementations of instances colliding with other things.
+// In this case, we treat instances as their polygons.
+////////////////////////////////////
+
+// Nabeel Danish
+
+#include "Universal_System/Object_Tiers/collisions_object.h"
+#include "Universal_System/Instances/instance_system.h" //iter
+#include "Universal_System/Instances/instance.h"
+#include "Universal_System/math_consts.h"
+#include "Universal_System/Resources/polygon.h"
+#include "Universal_System/Resources/polygon_internal.h"
+#include "../General/collisions_general.h"
+
+#include "Polygonimpl.h"
+#include "polygon_collision_util.h"
+#include <cmath>
+#include <utility>
+
+static bool precise_collision_single(int intersection_left, int intersection_right, int intersection_top, int intersection_bottom,
```

Comment:
Wait, does this code not already exist under the "Precise" collision system?

Reply:
It does, and this should be refactored into a single file that I tried to make in the collisions general folder.